src/hotspot/share/opto/replacednodes.cpp

```diff
@@ -144,20 +144,22 @@ void ReplacedNodes::apply(Compile* C, Node* ctl) {
       work.clear();
       enqueue_use(initial, use, work);
       bool replace = true;
-      // Check that this use is dominated by ctl. Go ahead with the
-      // replacement if it is.
+      // Check that this use is dominated by ctl. Go ahead with the replacement if it is.
       while (work.size() != 0 && replace) {
         Node* n = work.pop();
         if (use->outcnt() == 0) {
           continue;
         }
         if (n->is_CFG() || (n->in(0) != NULL && !n->in(0)->is_top())) {
-          int depth = 0;
-          Node *m = n;
+          // Skip projections, since some of the multi nodes aren't CFG (e.g., LoadStore and SCMemProj).
+          if (n->is_Proj()) {
+            n = n->in(0);
+          }
```

Comment:
I would suggest to merge this with the if below and adjust the comment to something like:
`// Skip non-CFG nodes. Also skip projections, since some of the multi nodes aren't CFG (e.g., LoadStore and SCMemProj).`

Reply:
I considered that, but it would turn it into a while loop. Do you think it is worth it?

Reply:
Ah right. No, that's probably not worth it.